Library/Homebrew/cmd/upgrade.rb

```diff
@@ -3,6 +3,14 @@
 
 module Homebrew
   def upgrade
+    if !MacOS.can_build?
```

Comment:
Similarly, this also calls `FormulaInstaller`; is the intention for this to bail out earlier than that point?